### PR TITLE
Check if libatomic can be linked

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

If we're building on MacOS, do not link with libatomic. Tries to solve the error `ld: library not found for -latomic`

##### Test Plan

We need some test builds on Mac both on x86 and arm if possible.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by MrZammler** [comment]
Discussion during latest comments at #12366

**Review by vkalintiris — DISMISSED** [review]
LGTM, I still think it would be better to just revert https://github.com/netdata/netdata/pull/12366 for this release. However, if @Ferroin's okay with this PR we can go ahead and merge it.

**Comment by underhood** [comment]
> LGTM, I still think it would be better to just revert #12366 for this release. However, if @Ferroin's okay with this PR we can go ahead and merge it.

Wasn't there in the past a check trying to actually build some bit of code to see if -latomic is needed?

Anyway we can link it always but only if present in system then.

File: configure.ac [hunk 1]
```diff
@@ -1531,12 +1531,14 @@ if test "${enable_exporting_kinesis}" = "yes" -o \
...
 fi
 
-# Unconditionally link with the libatomic since:
+# Unconditionally link with the libatomic except on macos since:
 # - if atomics are not needed, the library will be unused
 # - if atomics are needed, either atomics are fulfilled by the compiler builtins
 #   and the library is unused or the library is needed anyway.
-OPTIONAL_ATOMIC_LIBS="-latomic"
-AC_SUBST([OPTIONAL_ATOMIC_LIBS])
+if test "${build_target}" != "macos"; then
```
Comment (underhood) [inline configure.ac:1538]: Should this be based on `AC_CHECK_LIB` instead of `build_target`?

Comment (MrZammler) [reply configure.ac:1538]: I don't think there's even libatomic in MacOS. Searched brew, there's only libatomic_ops, but not sure if it's the same thing....

Comment (underhood) [reply configure.ac:1538]: yes but wouldnt that make the fix universal for all various random places where libatomic is not available. Maybe 2nd hand apple logo branded computers are not the only such system in the world.

Comment (MrZammler) [reply configure.ac:1538]: I think the whole point of #12366 was to link the library always, and let the compiler decide. Most likely we would have some trouble already if it caused compiling issues on other platforms too...

Comment (underhood) [reply configure.ac:1538]: well we can merge it as it is as long as we take care of this after the release

Comment (MrZammler) [reply configure.ac:1538]: I can have a go at making this work right, but not sure if I can make it today...

**Comment by underhood** [comment]
as per Slack discussion I am wondering what triggers this as I had Netdata agent built and connected to cloud after https://github.com/netdata/netdata/pull/12554 without seeing this error

**Comment by underhood** [comment]
> as per Slack discussion I am wondering what triggers this as I had Netdata agent built and connected to cloud after #12554 without seeing this error (scaleway Mac M1 mini), i think there is a bit of mystery still before I hit the approve button

LOL @vkalintiris called it above.  The https://github.com/netdata/netdata/pull/12366 got merged just 2 days ago and this is what broke it.

**Comment by MrZammler** [comment]
Ok, I've pushed a commit, it basically reverts the check in configure.ac for libatomic, but it now checks always, instead of only if certain conditions are met.

Any opinions are welcome.

**Comment by underhood** [comment]
testing - fixes the build on Linux without libatomic

**Review by underhood — APPROVED** [review]

**Review by vlvkobal — APPROVED** [review]

**Comment by MrZammler** [comment]
> probably breaks `riscv64` but that has to be fixed properly in future (in a way that doesn't break libatomic less systems)

Yes,and seems interesting, I will try to have a go.

Thanks guys!

**Pull request merged**